src/crimson/osd/osdmap_gate.cc

```diff
@@ -54,6 +54,11 @@ seastar::future<epoch_t> OSDMapGate<OSDMapGateTypeV>::wait_for_map(
 
 template <OSDMapGateType OSDMapGateTypeV>
 void OSDMapGate<OSDMapGateTypeV>::got_map(epoch_t epoch) {
+  if (epoch < current) {
+    logger().info("got_map({}) < current({}), ignoring", epoch, current);
+    return;
+  }
+  assert(epoch == current + 1);
```

Comment:
The assertion failures are addressed by https://github.com/ceph/ceph/pull/48036.